samples/authUserFlow.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+function main(projectId, clientId, clientSecret, refreshToken) {
+  // [START bigquery_auth_user_flow]
+  const {UserRefreshClient} = require('google-auth-library');
+
+  /**
+   * TODO(developer): Uncomment the following lines before running the sample.
+   */
+  // const clientId = 'CLIENT ID'
+  // const clientSecret = 'CLIENT SECRET'
+  // const refreshToken = 'REFRESH TOKEN'
+  // const projectId = 'PROJECT ID'
+  const credentials = {
+    type: 'authorized_user',
+    client_id: clientId,
+    client_secret: clientSecret,
+    refresh_token: refreshToken,
+  };
+
+  const refreshClient = new UserRefreshClient();
```

Comment:
Is this not automatically done if you pass in the `credentials` object above as:

```js
new BigQuery({credentials});
```

?

Reply:
@stephenplusplus yep, this totally works.